### PR TITLE
Add support for 64 bit base unit types for expanded power support

### PR DESCRIPTION
**Pull request by phlptp**
Add compile time option for using a 64 bit base type in the underlying data structure to allow a significantly higher power count that the 4 byte type supports.    If an 8 byte type is given for UNITS_BASE_TYPE

for example in CMAKE set -DUNITS_BASE_TYPE=uint64_t  will use the base as an 8 byte type instead of the default 4.  If this is done the size restrictions and checks are turned off.    only 4 and 8 byte types are supported and only the size matters so for the purposes of this modification uint32_t and int32_t and float are all equivalent.

### CODE REVIEW

**Comment by phlptp** [comment]
This is in support of Issue #127

**Comment by phlptp** [comment]
@SimonHeybrock  I added compile time support for a wider integer base,  Can you take a look see if this would meet your needs.

**Comment by SimonHeybrock** [comment]
Thanks a lot, this looks promising! I will have a closer look when I am back to the office after the easter holidays.

File: units/CMakeLists.txt [hunk 1]
```diff
@@ -25,14 +25,20 @@ if(UNITS_HEADER_ONLY)
             units-header-only INTERFACE -DUNITS_NAMESPACE=${UNITS_NAMESPACE}
         )
     endif()
+    if(UNITS_BASE_TYPE)
+        target_compile_definitions(
+            units-header-only PUBLIC -DUNITS_BASE_TYPE=${UNITS_BASE_TYPE}
```
Comment (SimonHeybrock) [inline units/CMakeLists.txt:30]: Is there a reason for exposing the type used in the bitfield via this flag? How about, e.g.,
- `cmake -DUNITS_BASE_TYPE_BYTES=4` (or 8) or
- `cmake -DUNITS_WIDE_BASE_TYPE=On`?

**Review by SimonHeybrock — COMMENTED** [review]
Seems to work, except for `sqrt`. It looks like there are some hard-coded `0` bits in `unit_data::root()` for base units such as `mole` and `counts` which need updating.

File: units/CMakeLists.txt (hunk 1 above)
Comment (phlptp) [reply units/CMakeLists.txt:30]: The main reason was consistency with some other libraries I interact with.    It isn't using the actual type you give it just the size of it, just the size but it does allow type operations on the define, whereas using the number would have to be translated.  Not that big of deal to do it differently but this way simplified a few implementation details slightly and was consistent with some other libraries.    So not a particularly strong reason.

**Comment by phlptp** [comment]
I will take a look at the root operation shortly

**Comment by SimonHeybrock** [comment]
Reran our tests, `sqrt` works now as expected 👍

**Comment by phlptp** [comment]
What were you using as a test that failed @SimonHeybrock?  I should probably add it or something like it to the test suite.

**Comment by SimonHeybrock** [comment]
> What were you using as a test that failed @SimonHeybrock? I should probably add it or something like it to the test suite.

`sqrt(mole * mole)` and `sqrt(counts * counts)`.

**Pull request merged**